src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
+  }
```

Comment:
[CODE PERFOMANCE] - The method `checkHealth` uses an inefficient way of removing dead animals from the `Animal.alive` array by using `indexOf` and `splice`. It is preferred to use the `filter` method to handle this operation to avoid nesting loops and improve performance.